pkg/controllers/iamauthpolicy_controller.go

```diff
@@ -109,10 +114,10 @@ func (c *IAMAuthPolicyController) Reconcile(ctx context.Context, req ctrl.Reques
 }
 
 func (c *IAMAuthPolicyController) reconcileDelete(ctx context.Context, k8sPolicy *anv1alpha1.IAMAuthPolicy) (ctrl.Result, error) {
-	err := c.validateSpec(ctx, k8sPolicy)
+	err := c.ph.ValidateTargetRef(ctx, k8sPolicy)
 	if err == nil {
 		modelPolicy := model.NewIAMAuthPolicy(k8sPolicy)
-		_, err := c.policyMgr.Delete(ctx, modelPolicy)
+		_, err := c.pm.Delete(ctx, modelPolicy)
 		if err != nil {
 			return ctrl.Result{}, services.IgnoreNotFound(err)
 		}
```

Comment:
For this function:
```
func (c *IAMAuthPolicyController) reconcileDelete(ctx context.Context, k8sPolicy *anv1alpha1.IAMAuthPolicy) (ctrl.Result, error) {
	err := c.ph.ValidateTargetRef(ctx, k8sPolicy)
	if err == nil {
		modelPolicy := model.NewIAMAuthPolicy(k8sPolicy)
		_, err := c.pm.Delete(ctx, modelPolicy)
		if err != nil {
			return ctrl.Result{}, services.IgnoreNotFound(err)
		}
	}
	err = c.handleLatticeResourceChange(ctx, k8sPolicy, model.IAMAuthPolicyStatus{})
	if err != nil {
		return ctrl.Result{}, err
	}
	c.removeFinalizer(k8sPolicy)
	return ctrl.Result{}, nil
}

```
Seem this function itself do a `c.pm.Delete()` and within ` c.handleLatticeResourceChange()`, it do a `c.pm.Delete()` again, is that expected?


Do you mind to write some code comments, or PR comments on how we handle deleting event for happyPathAuthPolicy, targetRefNotFoundPolicy and conflictedPolicy(i.e, didn't rank 1st after the `conflictResolutionSort` )?  This part logic is not really straightforward to me.


(I have similar question on `reconcileUpsert()`, seems we do `c.pm.Put()` first, then do a conditional `c.pm.Delete()`in `handleLatticeResourceChange()`, also expected?)